### DISCUSSION TITLE
Renovate doesn't work in a repository which has many dependencies

### DISCUSSION THREAD

**Discussion by suzuki-shunsuke** [renovatebot/renovate#21410] [Request Help]
### I'm using the Mend Renovate hosted app on github.com.

- [X] I'm using the Mend Renovate hosted app.

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe your problem.

- Renovate doesn't work in a specific repository. https://github.com/aquaproj/aqua-registry
- Renovate hasn't created pull requests since Apr 5 2023 (5 days ago)
- No log is listed in Renovate Dashboard
  - So I can't check Renovate log

<img width="1437" alt="image" src="https://user-images.githubusercontent.com/13323303/230907973-89fc3e36-74c3-46b8-adac-faa54b1fdc51.png">

- Renovate worked normally until 5 days ago
- Renovate configuration: https://github.com/aquaproj/aqua-registry/blob/f7308abecdbefb97fd83a73e71d058f954bc9813/renovate.json5

**This repository has many (over 1100) dependencies, so maybe this is related.**

### Relevant debug logs.

As I mentioned above, I can't check Renovate log.
But many `Dependency Lookup Warnings` is outputted in Dependency Dashboard.

https://github.com/aquaproj/aqua-registry/issues/1#issue-986081707

> ### ⚠ Dependency Lookup Warnings ⚠
> 
> -   Renovate failed to look up the following dependencies: `Failed to look up github-releases package bridgecrewio/yor`, `Failed to look up github-releases package kubernetes/kubectl-convert`, `Failed to look up github-releases package yannh/kubeconform`, `Failed to look up github-releases package ycd/dstp`, `Failed to look up github-releases package yinheli/sshw`, `Failed to look up github-releases package yitsushi/totp-cli`, `Failed to look up github-releases package yohamta/dagu`, `Failed to look up github-releases package yoheimuta/protolint`, `Failed to look up github-releases package ysugimoto/falco`, `Failed to look up github-releases package yudai/gotty`, `Failed to look up github-releases package zaquestion/lab`, `Failed to look up github-releases package zegl/kube-score`, `Failed to look up github-releases package zellij-org/zellij`, `Failed to look up github-releases package ziglang/zig`, `Failed to look up github-releases package zigtools/zls`, `Failed to look up github-releases package zix99/rare`, `Failed to look up github-releases package zquestz/s`, `Failed to look up github-releases package zurawiki/gptcommit`, `Failed to look up github-releases package zyedidia/eget`, `Failed to look up github-releases package zyedidia/micro`, `Failed to look up github-releases package oven-sh/bun`, `Failed to look up github-tags package golang/go`, `Failed to look up github-releases package ipinfo/cli`, `Failed to look up github-releases package kubernetes/kubernetes`, `Failed to look up github-tags package kubernetes/kubectl`, `Failed to look up github-releases package kubernetes-sigs/kustomize`, `Failed to look up github-releases package mongodb/mongodb-atlas-cli`, `Failed to look up github-releases package orf/gping`, `Failed to look up github-releases package grpc/grpc-go`.
> 
> Files affected: `pkgs/bridgecrewio/yor/pkg.yaml`, `pkgs/kubernetes/kubectl-convert/pkg.yaml`, `pkgs/yannh/kubeconform/pkg.yaml`, `pkgs/ycd/dstp/pkg.yaml`, `pkgs/yinheli/sshw/pkg.yaml`, `pkgs/yitsushi/totp-cli/pkg.yaml`, `pkgs/yohamta/dagu/pkg.yaml`, `pkgs/yoheimuta/protolint/pkg.yaml`, `pkgs/ysugimoto/falco/pkg.yaml`, `pkgs/yudai/gotty/pkg.yaml`, `pkgs/zaquestion/lab/pkg.yaml`, `pkgs/zegl/kube-score/pkg.yaml`, `pkgs/zellij-org/zellij/pkg.yaml`, `pkgs/ziglang/zig/pkg.yaml`, `pkgs/zigtools/zls/pkg.yaml`, `pkgs/zix99/rare/pkg.yaml`, `pkgs/zquestz/s/pkg.yaml`, `pkgs/zurawiki/gptcommit/pkg.yaml`, `pkgs/zyedidia/eget/pkg.yaml`, `pkgs/zyedidia/micro/pkg.yaml`, `pkgs/oven-sh/bun/pkg.yaml`, `pkgs/golang/go/pkg.yaml`, `pkgs/ipinfo/cli/pkg.yaml`, `pkgs/ipinfo/cli/cidr2ip/pkg.yaml`, `pkgs/ipinfo/cli/cidr2range/pkg.yaml`, `pkgs/ipinfo/cli/range2cidr/pkg.yaml`, `pkgs/ipinfo/cli/prips/pkg.yaml`, `pkgs/ipinfo/cli/splitcidr/pkg.yaml`, `pkgs/ipinfo/cli/randip/pkg.yaml`, `pkgs/ipinfo/cli/grepip/pkg.yaml`, `pkgs/ipinfo/cli/range2ip/pkg.yaml`, `pkgs/kubernetes/kubectl/pkg.yaml`, `pkgs/kubernetes-sigs/kustomize/pkg.yaml`, `pkgs/mongodb/mongodb-atlas-cli/atlascli/pkg.yaml`, `pkgs/orf/gping/pkg.yaml`, `pkgs/grpc/grpc-go/protoc-gen-go-grpc/pkg.yaml`

**Comment by suzuki-shunsuke** [comment]
I separated dependencies with [schedule](https://docs.renovatebot.com/configuration-options/#schedule).
Then Renovate was recovered.

- https://github.com/aquaproj/aqua-registry/pull/11371

```json5
  packageRules: [
    {
      matchPackagePatterns: ["^[a-k]"],
      schedule: ["before 12pm"],
    },
    {
      matchPackagePatterns: ["^[^a-k]"],
      schedule: ["after 12pm"],
    },
  ],
```

**Reply by rarkins** [reply]
Thanks for updating. In that case it sounds like the job was taking too long and getting timed out by the app's backend. By splitting schedules, I guess that saves how many commits are made per run? Wondering if there's a better way to achieve it, such as hourly limits, or commits per run, etc.

**Reply by suzuki-shunsuke** [reply]
Renovate log. https://app.renovatebot.com/dashboard#github/aquaproj/aqua-registry/1046347899

```
INFO: Dependency extraction complete
{
  "baseBranch": "main",
  "stats": {
    "managers": {
      "github-actions": {
        "fileCount": 2,
        "depCount": 22
      },
      "regex": {
        "fileCount": 1123,
        "depCount": 1140
      }
    },
    "total": {
      "fileCount": 1125,
      "depCount": 1162
    }
  }
}
```

```
DEBUG: Package lookup durations
{
  "github-releases": {
    "count": 1134,
    "averageMs": 631,
    "totalMs": 715385,
    "maximumMs": 19269
  },
  "github-tags": {
    "count": 24,
    "averageMs": 1390,
    "totalMs": 33353,
    "maximumMs": 4865
  },
  "go": {
    "count": 2,
    "averageMs": 92,
    "totalMs": 183,
    "maximumMs": 143
  }
}
```
